tests/system/test_system.py

```diff
@@ -142,32 +145,121 @@ def tearDown(self):
     def _logger_name(prefix):
         return prefix + unique_resource_id("-")
 
-    def test_list_entry_with_unregistered(self):
-        from google.protobuf import any_pb2
+    @staticmethod
+    def _to_value(data):
+        if data is None:
+            return Value(null_value=NullValue.NULL_VALUE)
+        elif isinstance(data, numbers.Number):
+            return Value(number_value=data)
+        elif isinstance(data, str):
+            return Value(string_value=data)
+        elif isinstance(data, bool):
+            return Value(bool_value=data)
+        elif isinstance(data, (list, tuple, set)):
+            return Value(
+                list_value=ListValue(values=(TestLogging._to_value(e) for e in data))
+            )
+        elif isinstance(data, dict):
+            return Value(struct_value=TestLogging._dict_to_struct(data))
+        else:
+            raise TypeError("Unknown data type: %r" % type(data))
+
+    @staticmethod
+    def _dict_to_struct(data):
+        return Struct(fields={k: TestLogging._to_value(v) for k, v in data.items()})
+
+    def test_list_entry_with_auditlog(self):
         from google.protobuf import descriptor_pool
         from google.cloud.logging_v2 import entries
 
         pool = descriptor_pool.Default()
         type_name = "google.cloud.audit.AuditLog"
-        # Make sure the descriptor is not known in the registry.
-        with self.assertRaises(KeyError):
-            pool.FindMessageTypeByName(type_name)
+        type_url = "type.googleapis.com/" + type_name
+        # Make sure the descriptor is known in the registry.
+        # Raises KeyError if unknown
+        pool.FindMessageTypeByName(type_name)
+
+        # create log
+        audit_dict = {
+            "@type": type_url,
+            "methodName": "test",
+            "requestMetadata": {"callerIp": "::1", "callerSuppliedUserAgent": "test"},
+            "resourceName": "test",
+            "serviceName": "test",
+            "status": {"code": 0},
+        }
+        audit_struct = self._dict_to_struct(audit_dict)
+
+        logger = Config.CLIENT.logger("audit-proto")
+        logger.log_proto(audit_struct)
+
+        # retrieve log
+        filter_ = self.TYPE_FILTER.format(type_url) + f" AND {_time_filter}"
```

Comment:
this is a side note for logging language syntax and not to this specific line:
I find it ugly when we concatenate strings to form a query. Perhaps, we should think about creation of an abstraction that will make forming these filters more intuitive when you read the code.
For example:

Instead of 
```
xxx + " AND " + time_filter
```
it could be 
```
x.And(new BinaryOperator(time1, operands.Equal, time2)
```
This could be a separate task and not part of this PR

Reply:
That's an interesting idea. I'm not sure if that specific example would be seen as idiomatic for Python, but there's probably something interesting we could do with this

Reply:
2 cents: I recall in Python (and Node), this kind of interpolation is preferred